crates/ruff_python_formatter/src/lib.rs

```diff
@@ -50,10 +51,14 @@ where
 
         let node_comments = comments.leading_dangling_trailing_comments(node.as_any_node_ref());
 
-        leading_comments(node_comments.leading).fmt(f)?;
-        self.fmt_node(node, f)?;
-        self.fmt_dangling_comments(node_comments.dangling, f)?;
-        trailing_comments(node_comments.trailing).fmt(f)
+        if self.is_suppressed(node_comments.trailing, f.context()) {
```

Comment:
I considered handling the suppression in the `FormatStmt` so that it isn't necessary to override `is_suppressed` for all statements. However, it requires that `FormatStmt` resolves the trailing comments to test for a suppression comment, just for the `FormatNodeRule` to lookup the very same comments again.